Jim dev react

jsharples777 opened the pull request:
Much tidier
Working dynamic pagination now
Removed high fibre, EDAMAM don't accept it despite it being in their documentation

The presentation HTML is now mostly in the JSX files, which is where we will need to work to fix up the UI glitches.

Outstanding:
1.  Search button
2. User feedback when search is running, and prevent second search request while it is still running
3. Cards for results seem too small and image sizes vary creating a mixed look when looking at results
4. Need icons with the + sign for the add to favourites and shopping list
5. Overall look tweaking
6. Are we going to link to the Google Map API?

jsharples777 committed:
Converted to React
Created dynamic pagination based on number of results

jsharples777 committed:
Removed high-fiber, despite API document, doesn't accept as criteria

jsharples777 commented:
Also, once installed:
1.  re-run npm install
2.  need two terminal windows, 
     a) npm run start - for the webserver to run on localhost:3000
     b) npm run watch - which will dynamically convert the JSX to JS files as they are changed and saved, to play with changing look and feel and style classes

jsharples777 committed:
New star plus ixon

raven-bootcamp commented:
> 1. Search button
> 2. User feedback when search is running, and prevent second search request while it is still running
> 3. Cards for results seem too small and image sizes vary creating a mixed look when looking at results
> 4. Need icons with the + sign for the add to favourites and shopping list
> 5. Overall look tweaking
> 6. Are we going to link to the Google Map API?

1, 2 and 4 : I'll create cards for these.
3 and 5 : I think Irene is working on them by the sounds of it.
6: We decided against that, it was more of a stretch goal if we had time, but it seemed like a lot of extra work.

jsharples777 committed:
1.  UI tweaks
2.  Image fixed sizing
3.  Tidier Recipe details
4.  Clear filters button on search
5.  Search with loading button when running and won't accept new searches
6.  Fixed very long named items to be shorter and ... on end of names

jsharples777 committed:
Fixed shopping list bug
Getting ready for deployment

jsharples777 committed:
Fixing mobile alignment issues

jsharples777 committed:
UI tweaks and style changes (per collaboration with Irene)

jsharples777 committed:
Merge branch 'main' into jim_dev_react

jsharples777 committed:
Start location API

jsharples777 committed:
Merge remote-tracking branch 'origin/jim_dev_react' into jim_dev_react
